subprojects/model-core/src/main/java/org/gradle/api/internal/provider/AbsentProviderHandling.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.provider;
+
+import org.gradle.api.provider.Provider;
+
+import javax.annotation.Nullable;
+
+public enum AbsentProviderHandling {
+    ALLOW_ABSENT {
+        @Override
+        public <T> T getValue(Provider<T> provider) {
+            return provider.getOrNull();
+        }
+    },
+    REQUIRE_PRESENT {
+        @Override
+        public <T> T getValue(Provider<T> provider) {
+            return provider.get();
+        }
+    };
+
+    @Nullable
+    public abstract <T> T getValue(Provider<T> provider);
```

Comment:
We might consider calling this `resolveValue()`, or just `resolve()`. I know we use that word in many places, but it fits well what we do here I think.

Reply:
Renamed.